manager/allocator/networkallocator/networkallocator.go

```diff
@@ -878,3 +890,16 @@ func serviceNetworks(s *api.Service) []*api.NetworkAttachmentConfig {
 	}
 	return s.Spec.Task.Networks
 }
+
+//IsVipOnIngressNetwork check if the vip is in ingress network
+func (na *NetworkAllocator) IsVipOnIngressNetwork(vip *api.Endpoint_VirtualIP) bool {
+	if vip == nil {
+		return false
+	}
+
+	localNet := na.getNetwork(vip.NetworkID)
+	if localNet != nil && localNet.nw != nil {
+		return localNet.nw.Spec.Ingress
```

Comment:
@aaronlehmann I agree it is not ideal.

@abhinandanpb  As @aaronlehmann  commented, the ingress check has to be complete, it has to also check for the legacy way of expressing a network as ingress.

IMO, we should have only one method run the check. 
We can refactor the double methods later IMO. Given this is fixing an imminent regression problem.